score/score_test.go

```diff
@@ -0,0 +1,66 @@
+package score
+
+import "testing"
+
+var theSecret = "my secret code"
+
+func init() {
+	GlobalSecret = theSecret
+}
+
+var nonJSONLog = []string{
+	"heere is some output",
+	"some other output",
+	"line contains " + theSecret,
+	theSecret + " should not be revealed",
+}
+
+func TestParseNonJSONStrings(t *testing.T) {
+	for _, s := range nonJSONLog {
+		sc, err := Parse(s, GlobalSecret)
+		if err == nil {
+			t.Errorf("Expected '%v', got '<nil>'", ErrScoreNotFound.Error())
+		}
+		if sc != nil {
+			t.Errorf("Got unexpected score object '%v', wanted '<nil>'", sc)
+		}
+	}
+}
+
+var jsonLog = []struct {
+	in  string
+	out *Score
+	err error
+}{
+	{`{"Secret":"` + theSecret + `","TestName":"init","Score":0,"MaxScore":10,"Weight":10}`,
+		NewScore(10, 10),
+		nil,
+	},
+	{`{"Secret":"the wrong secret","TestName":"init","Score":0,"MaxScore":10,"Weight":10}`,
+		nil,
+		ErrScoreNotFound,
+	},
+}
+
+// Equal returns true if sc equals other. Ignores the Secret field.
```

Comment:
This library is a helper library for everyone who wants to write tests(or similar) for use with AG, so I think it would be more useful to have it in the main struct definition since it's already implemented. Then people have the option to use it.